zilliqa/src/api/ots.rs

```diff
@@ -234,3 +237,42 @@ fn search_transactions_before(
 
     search_transactions_inner(node, Address(address), block_number, page_size, true)
 }
+
+fn get_internal_operations(
+    params: Params,
+    node: &Arc<Mutex<Node>>,
+) -> Result<Vec<InternalOperationOtter>> {
+    trace!("get_internal_operations called");
+    let hash: H256 = params.one()?;
+    let hash: Hash = Hash(hash.0);
+
+    let node = node.lock().unwrap();
+
+    let tx = node.get_transaction_by_hash(hash)?;
+
+    let receipt = get_transaction_receipt_inner(hash, &node)?;
+
+    if tx.is_none() || receipt.is_none() {
+        return Err(anyhow!("transaction not yet executed: {hash}"));
+    }
+
+    let tx = tx.unwrap();
+    let receipt = receipt.unwrap();
+
+    let call_result = node.call_contract(
+        receipt.block_number.into(),
+        tx.from_addr,
+        tx.transaction.to_addr,
+        tx.transaction.payload,
+        tx.transaction.amount.into(),
+        true,
+    );
+
+    Ok(call_result
+        .unwrap()
+        .tx_trace
+        .lock()
+        .unwrap()
+        .otter_internal_tracer
```

Comment:
Oh, I would have kept it where it is as I don't like globals too much so seems cleaner to keep it inside the evm. We can move it out if that's your preference.

IIRC it was originally created where it was because the SputnikEVM didn't support a single listener that would capture all of the info we needed (so you would have had to create nested listeners).